OpenRA.Game/Activities/Activity.cs

```diff
@@ -40,6 +40,15 @@ public virtual IEnumerable<Target> GetTargets(Actor self)
 		{
 			yield break;
 		}
+
+		public virtual CPos GetDestination(Actor self)
+		{
+			return CPos.Null;
+		}
```

Comment:
Make this into a public abstract bool? Destination { get; } or public virtual bool? Destination { get { return null; } }, this allows you to drop the change to CPos and prevent a possible bug if some code uses -1, -1 as an actual destination (as in go outside the map) also allows you to simply do a Destination.HasValue for a null check.